gateway-provider-rewrite/src/main/java/org/apache/knox/gateway/filter/rewrite/impl/html/HtmlFilterReaderBase.java

```diff
@@ -288,19 +279,13 @@ private QName getQName( String name ) {
     }
 
     private void parseNamespaces() {
-      Attributes attributes = tag.getAttributes();
-      if( attributes != null ) {
-        for( Attribute attribute : tag.getAttributes() ) {
-          String name = attribute.getName();
-          if( name.toLowerCase(Locale.ROOT).startsWith( "xmlns" ) ) {
-            int colon = name.indexOf( ':', 5 );
-            String prefix;
-            if( colon <= 0 ) {
-              prefix = "";
-            } else {
-              prefix = name.substring( colon );
-            }
-            namespaces.put( prefix, attribute.getValue() );
+      if (tag.getAttributes() != null) {
+        String prefix;
+        for (Attribute attribute : tag.getAttributes()) {
+          if (attribute.getName().toLowerCase(Locale.ROOT).startsWith("xmlns")) {
```

Comment:
Might help to pull `attribute.getName()` out to a local variable since it is used a bunch of times.